### PR TITLE
Added frontend dependency management with Grunt + Bower + NodeJS. Various Dashboard fixes.

### PR DESCRIPTION
**Pull request by adriannieto**
This allows to have automatic dependency resolution at compile time. It downloads all the required libraries noted by bower and packages into a single minified file.  This simplifies the development of the dashboard, and reduces the overhead of the project on Github. This addition is totally transparent for the user. 

If this PR is merged, EVERY contribution to the Dashboard module MUST use bower to handle the dependencies.

Note: This PR adds nodejs, bower and grunt as a requirement to build the Dashboard. 

@andreaturli @rosogon could you review this? @andreaturli should I add something to travis to install Node,Bower and Grunt in order to make this to build automatically?

### CODE REVIEW

File: dashboard/src/main/resources/webapp/static/lib/angular-topology-editor/canvas.js [hunk 1]
```diff
@@ -101,7 +101,8 @@ Graph.Node.popovercontent = function(i) {
 };
 
 var Canvas = (function() {
-
+    var LOG = Log.getLogger("Canvas").setLevel(Log.INFO);
```
Comment (rosogon) [inline dashboard/src/main/resources/webapp/static/lib/angular-topology-editor/canvas.js]: Why this change? loglevel.js is better than log.js

Comment (adriannieto) [inline dashboard/src/main/resources/webapp/static/lib/angular-topology-editor/canvas.js]: log was defined as a global variable, and I had two choices because two log libraries did exist. I took this one because it felt more simple to me.

Comment (rosogon) [inline dashboard/src/main/resources/webapp/static/lib/angular-topology-editor/canvas.js]: log.js is the library I wrote, but it has a big disadvantage: you don't know where was the log statement. That's why I switched to loglevel. I should have removed log.js.

Regarding "log" being a global variable, we can overcome it with:
https://github.com/pimterry/loglevel#with-noconflict

Comment (adriannieto) [inline dashboard/src/main/resources/webapp/static/lib/angular-topology-editor/canvas.js]: Because the way AngularJS works I think that probably you can't retrieve in your scripts the variable from loglevel.js as they do in the example:

```
<script src="loglevel.min.js"></script>
<script>
var logging = log.noConflict();

logging.error("still pretty easy");
</script>
```

 I really discourage this kind of implementations that loads global variables in conjunction with AngularJS, it could lead into serious conflicts.

Comment (rosogon) [inline dashboard/src/main/resources/webapp/static/lib/angular-topology-editor/canvas.js]: @adriannieto, all js libraries export variables as global to access their code:

* jquery exports $ and jQuery
* d3 exports d3
* our canvas, editor, etc, export Canvas, Editor...

loglevel does the same, and even gives the option to choose the global name you need to avoid conflicts.

Comment (adriannieto) [inline dashboard/src/main/resources/webapp/static/lib/angular-topology-editor/canvas.js]: I know, but logging is IMO a minor feature heavily used across libraries, anyway if you want I could try to change it back.

**Comment by rosogon** [comment]
Is it possible to install bower in the build? 

```
[INFO] OS Name: Windows 7
'bower' is not recognized as an internal or external command,
operable program or batch file.
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO]
[INFO] SeaClouds Platform ................................. SUCCESS [ 17.116 s]
[INFO] common ............................................. SUCCESS [01:30 min]
[INFO] deployer ........................................... SUCCESS [01:00 min]
[INFO] planner ............................................ SUCCESS [  2.464 s]
[INFO] matchmaker ......................................... SUCCESS [ 33.025 s]
[INFO] optimizer .......................................... SUCCESS [ 13.673 s]
[INFO] optimizer-core ..................................... SUCCESS [ 14.385 s]
[INFO] optimizer-service .................................. SUCCESS [01:38 min]
[INFO] planner-core ....................................... SUCCESS [  4.939 s]
[INFO] planner-service .................................... SUCCESS [ 29.035 s]
[INFO] aamwriter .......................................... SUCCESS [ 26.979 s]
[INFO] discoverer ......................................... SUCCESS [ 31.349 s]
[INFO] monitor ............................................ SUCCESS [  0.310 s]
[INFO] nuro-data-collector ................................ SUCCESS [ 57.312 s]
[INFO] reconfiguration-data-collector ..................... SUCCESS [ 11.218 s]
[INFO] Dashboard .......................................... FAILURE [06:30 min]
[INFO] SLA ................................................ SKIPPED
[INFO] SLA core parent .................................... SKIPPED
[INFO] SLA repository database ............................ SKIPPED
[INFO] SLA Enforcement .................................... SKIPPED
[INFO] SLA Tools .......................................... SKIPPED
[INFO] SLA Project Personalization ........................ SKIPPED
[INFO] SLA Service ........................................ SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 14:51 min
[INFO] Finished at: 2015-10-26T12:49:26+00:00
[INFO] Final Memory: 106M/658M
[INFO] ------------------------------------------------------------------------
```

We don't have a section for "Compiling SeaClouds" in the README. Maybe we should add one (not to be covered by this PR).

**Comment by andreaturli** [comment]
this will never work on CI server, as-is.

**Comment by adriannieto** [comment]
@andreaturli what do you suggest? Currently it's building but I think that installing node, bower or grunt on the fly it's not the best way

**Comment by andreaturli** [comment]
@adriannieto so better not to build the dashboard, I guess, right?

**Comment by rosogon** [comment]
@adriannieto , @andreaturli . The build needs nodejs, bower and grunt, but travis is building the project. 

I thought it better, and better not to install bower on the fly.

**Comment by andreaturli** [comment]
@adriannieto, looking at https://travis-ci.org/SeaCloudsEU/SeaCloudsPlatform/builds/87439909#L28
Travis contains node.js which should be the only prerequisite according to the [doc](https://github.com/allegro/grunt-maven-plugin#prerequisites) of the maven plugin you are using.
It doesn't seem to require bower also pre-installed.

**Comment by rosogon** [comment]
When trying to load a file in the last step... 
```
TypeError: Cannot read property '0' of undefined
    at Object.fn (add-application-wizard.js:340)
    at n.$digest (dependencies.js:13)
    at n.$apply (dependencies.js:13)
    at $$debounceViewValueCommit (dependencies.js:15)
    at $setViewValue (dependencies.js:15)
    at n (dependencies.js:29)
    at dependencies.js:29
    at dependencies.js:13
    at e (dependencies.js:11)
    at dependencies.js:11
```

**Comment by rosogon** [comment]
+1

@adriannieto : Can you squash in two different commits. One for the grunt+bower, and one for the fixes. Thanks

**Comment by adriannieto** [comment]
@rosogon Job done.
